src/app/services/xml-parsers/msdesc-parser.ts

```diff
@@ -0,0 +1,226 @@
+            attributes: this.attributeParser.parse(xml),
+            summary,
+            msItem,
+            msItemStruct,
+        };
+    }
+}
+
+export class MsIdentifierParser extends EmptyParser implements Parser<XMLElement> {
+    attributeParser = createParser(AttributeParser, this.genericParse);
+
+    parse(xml: XMLElement): MsIdentifier {
+        // TODO: Add specific parser when settlement is handled
+        const settlement = Array.from(xml.querySelectorAll<XMLElement>(':scope > settlement'))
+        .map(e => parseChildren(e, this.genericParse));
+        // TODO: Add specific parser when repository is handled
+        const repository = Array.from(xml.querySelectorAll<XMLElement>(':scope > repository'))
+        .map(e => parseChildren(e, this.genericParse));
+        // TODO: Add specific parser when idno is handled
+        const idno = Array.from(xml.querySelectorAll<XMLElement>(':scope > idno'))
+        .map(e => parseChildren(e, this.genericParse));
+        // TODO: Add specific parser when institution is handled
+        const institution = Array.from(xml.querySelectorAll<XMLElement>(':scope > institution'))
+        .map(e => parseChildren(e, this.genericParse));
+        // TODO: Add specific parser when altIdentifier is handled
+        const altIdentifier = Array.from(xml.querySelectorAll<XMLElement>(':scope > altIdentifier'))
+        .map(e => parseChildren(e, this.genericParse));
+        // TODO: Add specific parser when msName is handled
+        const msName = Array.from(xml.querySelectorAll<XMLElement>(':scope > msName'))
+        .map(e => parseChildren(e, this.genericParse));
+        // TODO: Add specific parser when msName is handled
+        const country = Array.from(xml.querySelectorAll<XMLElement>(':scope > country'))
+        .map(e => parseChildren(e, this.genericParse));
+        // TODO: Add specific parser when altIdentifier is handled
+        const region = Array.from(xml.querySelectorAll<XMLElement>(':scope > region'))
+        .map(e => parseChildren(e, this.genericParse));
+
+        return {
+            type: MsIdentifier,
+            class: getClass(xml),
+            content: parseChildren(xml, this.genericParse),
+            attributes: this.attributeParser.parse(xml),
+            settlement,
+            repository,
+            idno,
+            institution,
+            altIdentifier,
+            msName,
+            country,
+            region,
+        };
+    }
+}
+
+export class MsPartParser extends EmptyParser implements Parser<XMLElement> {
+    attributeParser = createParser(AttributeParser, this.genericParse);
+    private msIdentifierParser = createParser(MsIdentifierParser, this.genericParse);
+    private msContentsParser = createParser(MsContentsParser, this.genericParse);
+    private physDescParser = createParser(PhysDescParser, this.genericParse);
+    private msPartParser = createParser(MsPartParser, this.genericParse);
```

Comment:
You don't need this definition. You already are inside the `MsPartParser`. If you need to use it you can just do `this.parse()`;